BedrockServer.cpp

```diff
@@ -443,10 +445,19 @@ void BedrockServer::sync(const SData& args,
                 // state, because this loop is skipped except when LEADING, FOLLOWING, or STANDINGDOWN. It's also
                 // theoretically feasible for this to happen if a follower fails to commit a transaction, but that
                 // probably indicates a bug (or a follower disk failure).
-                SINFO("requeueing command " << command.request.methodLine
-                      << " after failed sync commit. Sync thread has " << syncNodeQueuedCommands.size()
-                      << " queued commands.");
-                syncNodeQueuedCommands.push(move(command));
+
+                // upgradeDB is known to be subject to this condition at cluster startup, if an unstable cluster state
+                // leads to high velocity leader flip-flopping before the upgrade has a chance to finish.
+                if (upgradeInProgress.load()) {
```

Comment:
I'm not sure this is the right fix, but will let @tylerkaraszewski comment.

Reply:
I think I'm OK with the fix, but I think this needs a better explanation. Something like:

While _upgradeDB isn't a normal command, it is still run as a QUORUM transaction, and so it can still fail if peers reject is (which they might do if it comes from a node that they don't think should currently be leading. In that case, we will fail an upgradeDB distributed transaction, and we need to reset that state.

Reply:
Got it. Updating the comment.